src/netlink.c

```diff
@@ -195,6 +195,8 @@ static int meth_receivefrom(lua_State *L) {
         return 2;
     }
 
+    got = got > NLMSG_SPACE(MAX_PAYLOAD) ? MAX_PAYLOAD : NLMSG_PAYLOAD(nlh, 0);
```

Comment:
I think you should create a new variable for this value, to make it clear that this is a different thing.